Behat/Context/UIContext.php

```diff
@@ -206,6 +206,52 @@ public function iClickOnTheLabel($label) {
     $this->iClickOnTheElementWithXPath($xpath);
   }
 
+  /**
+   * Get element by xpath.
+   */
+  protected function getElementByXpath($xpath_element) {
+    $page = $this->getSession()->getPage();
+    $findelement = $page->find('xpath', $xpath_element);
+    return $findelement;
+  }
+
+  /**
+   * Check if a type of element has an attribute with an specific value.
+   *
+   * @Then the :element element of :type type should have the :attribute attribute with :value value
+   */
+  public function theElementShouldHaveAttributeWithValue($element, $type, $attribute, $value, $not = FALSE) {
+    $not = is_null($not) ? FALSE : $not;
+    $xpath = "//{$type}[contains(text(),'{$element}')]";
+    $category_element = $this->getElementByXpath($xpath);
+
+    if (is_null($category_element)) {
+      throw new \Exception("The element {$element} was not found");
+    }
+
+    $xpath = $not ? "//{$type}[not(contains(@{$attribute},'{$value}'))][contains(text(),'{$element}')]" : "//{$type}[contains(@{$attribute},'{$value}')][contains(text(),'{$element}')]";
+
+    $category_element = $this->getElementByXpath($xpath);
+
+    if (is_null($category_element) && $not) {
+      print $xpath;
+      throw new \Exception("The element {$element} does not have the attribute {$attribute} with the value {$value}");
+    }
+    elseif (is_null($category_element) && !$not) {
+      print $xpath;
+      throw new \Exception("The element {$element} has the attribute {$attribute} with the value {$value}");
+    }
+  }
```

Comment:
I would rewrite previous block to:

```
if (is_null($category_element)) {
  print $xpath;
  $condition_error_string = $not ? "does not" : "has";
  throw new \Exception("The element {$element} ${condition_error_string} have the attribute {$attribute} with the value {$value}");
  }
```

I think this way makes clear that the error comes from the $category_element being null, and inside there's a condition to get the exact error message.

My last concern is the print statement. Why print the xpath expression? Is that a debug sentence?